src/bot.py

```diff
@@ -241,13 +241,21 @@ def on_maintenance(self):
         self._no_stash_counter += 1
         force_stash = self._no_stash_counter > 4 and self._ui_manager.should_stash(self._config.char["num_loot_columns"])
         if self._picked_up_items or force_stash:
-            Logger.info("Stashing items")
-            self._curr_loc = self._town_manager.stash(self._curr_loc)
-            if not self._curr_loc:
-                return self.trigger_or_stop("end_game", failed=True)
-            self._no_stash_counter = 0
-            self._picked_up_items = False
-            wait(1.0)
+            #if ID item is set, then let's id them before we stash them.
+            if self._config.char["ID_items"]:
+                Logger.info("Identifying items")
+                #do the iding stuff here.
+                self._curr_loc = self._town_manager.identify(self._curr_loc)
+                if not self._curr_loc:
+                    return self.trigger_or_stop("end_game", failed=True)         
+
+                Logger.info("Stashing items")
```

Comment:
This stashing block of code should be tabbed back a level and not included in the if IDing block. You still want to stash even if ID is turned off.

Reply:
missed that. thanks.